src/admin/index.js

```diff
@@ -368,12 +375,31 @@ module.exports = ({
     const coordinator = await cluster.findGroupCoordinator({ groupId })
     const partitionsToFetch = partitions.map(partition => ({ partition }))
 
-    const { responses } = await coordinator.offsetFetch({
+    let { responses: consumerOffsets } = await coordinator.offsetFetch({
       groupId,
       topics: [{ topic, partitions: partitionsToFetch }],
     })
 
-    return responses
+    if (resolveOffsets) {
+      const indexedOffsets = indexByPartition(await fetchTopicOffsets(topic))
+      consumerOffsets = consumerOffsets.map(({ topic, partitions }) => ({
+        topic,
+        partitions: partitions.map(({ offset, partition, ...props }) => ({
+          partition,
+          offset:
+            Number(offset) === EARLIEST_OFFSET
+              ? indexedOffsets[partition].low
+              : Number(offset) === LATEST_OFFSET
+              ? indexedOffsets[partition].high
+              : offset,
+          ...props,
+        })),
```

Comment:
```suggestion
        partitions: partitions.map(({ offset, partition, ...props }) => {
          let resolvedOffset = offset
          if (Number(offset) === EARLIEST_OFFSET) {
            resolvedOffset = indexedOffsets[partition].low
          }
          if (Number(offset) === LATEST_OFFSET) {
            resolvedOffset = indexedOffsets[partition].high
          }
          return {
            partition,
            offset: resolvedOffset,
            ...props,
          }
        }),
```

@tulios, checking before I commit, if this is what you had in mind to make it more readable, or you were thinking something different?

Reply:
Yes, the suggestion works fine.